plugins/woocommerce/includes/wc-update-functions.php

```diff
@@ -2730,6 +2730,45 @@ function wc_update_910_add_launch_your_store_tour_option() {
 	add_option( 'woocommerce_show_lys_tour', 'yes' );
 }
 
+/**
+ * Add woocommerce_hooked_blocks_version option for existing stores that are using a theme that supports the Block Hooks API
+ */
+function wc_update_920_add_wc_hooked_blocks_version_option() {
+	if ( ! wc_current_theme_is_fse_theme() && ! current_theme_supports( 'block-template-parts' ) ) {
+		return;
+	}
+
+	$option_name       = 'woocommerce_hooked_blocks_version';
+	$option_value      = get_option( $option_name );
+	$active_theme_name = wp_get_theme()->get( 'Name' );
+
+	// If the option already exists, we don't need to do anything.
+	if ( false !== $option_value ) {
+		return;
+	}
+
+	/**
+	 * A list of theme slugs to execute this with.
+	 * We are applying this filter to allow for the list to be extended by third-parties who were already using it.
+	 *
+	 * @since 8.4.0
+	 */
+	$theme_include_list               = apply_filters( 'woocommerce_hooked_blocks_theme_include_list', array( 'Twenty Twenty-Four', 'Twenty Twenty-Three', 'Twenty Twenty-Two', 'Tsubaki', 'Zaino', 'Thriving Artist', 'Amulet', 'Tazza' ) );
+	$should_set_hooked_blocks_version = in_array( $active_theme_name, $theme_include_list, true );
+
+	if ( $should_set_hooked_blocks_version && false === $option_value ) {
+		// Set 8.4.0 as the version for existing stores that are using a theme that supports the Block Hooks API.
+		// This will ensure that the Block Hooks API is enabled for these stores and works as expected.
+		// Existing stores that aren't running approved block themes will not have the Block Hooks API enabled.
+		add_option( $option_name, '8.4.0' );
+	} elseif ( ! $should_set_hooked_blocks_version && false === $option_value ) {
```

Comment:
It should be possible to remove `false === $option_value` from these checks now, no?

```suggestion
	if ( $should_set_hooked_blocks_version ) {
		// Set 8.4.0 as the version for existing stores that are using a theme that supports the Block Hooks API.
		// This will ensure that the Block Hooks API is enabled for these stores and works as expected.
		// Existing stores that aren't running approved block themes will not have the Block Hooks API enabled.
		add_option( $option_name, '8.4.0' );
	} elseif ( ! $should_set_hooked_blocks_version ) {
```

Reply:
Done in https://github.com/woocommerce/woocommerce/pull/49302/commits/8f583ee6714d2f839e832f28756d5402bef7c45a